src/CodeGen_Hexagon.cpp

```diff
@@ -501,22 +501,27 @@ llvm::Function *CodeGen_Hexagon::define_hvx_intrinsic(llvm::Function *intrin, Ty
     return wrapper;
 }
 
+Value *CodeGen_Hexagon::create_bitcast(Value *v, llvm::Type *ty) {
+    if (BitCastInst *c = dyn_cast<BitCastInst>(v)) {
+        return create_bitcast(c->getOperand(0), ty);
```

Comment:
I'm sure LLVM does this too, the motivation for this isn't to help out LLVM
generate better code from otherwise identical IR, it's to make it so we can
find vcombine and other instructions without having to peel away more than
one bitcast. That lets us generate better IR that LLVM currently can't do
itself.

On Sun, Apr 17, 2016 at 7:01 PM Pranav Bhandarkar <notifications@github.com>
wrote:

> In src/CodeGen_Hexagon.cpp
> <https://github.com/halide/Halide/pull/1129#discussion_r59999344>:
>
> > @@ -501,22 +501,27 @@ llvm::Function *CodeGen_Hexagon::define_hvx_intrinsic(llvm::Function *intrin, Ty
> >      return wrapper;
> >  }
> >
> > +Value *CodeGen_Hexagon::create_bitcast(Value *v, llvm::Type *ty) {
> > +    if (BitCastInst *c = dyn_cast<BitCastInst>(v)) {
> > +        return create_bitcast(c->getOperand(0), ty);
>
> This is strange. Isn't LLVM able to cancel out redundant bitcasts? Is this
> leading to sub-optimal code?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/1129/files/839bf5f0bdcab37fb5459bedd0d685c6b314e293#r59999344>
>